### PR TITLE
New print styles and generator to add to build

### PR DESCRIPTION
**Pull request by jathayde**
Screenshot 2024-03-31 at 8 43 35 PM

### Motivation / Background

This updates print CSS to be more inline with the overall style. The sidebar is inlined (same as mobile), and content is 100%. Previous code in the launch from boilerplate has been struck or integrated better.

* Links show in text after the original link text
* Border line around code and copy button are removed
* Print is now generated from SCSS to CSS via the generator that builds the other CSS files 

### Additional information

Some browsers will override backgrounds on print

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/assets/stylesrc/print.scss [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      background: url("../images/hr_rails.svg") center no-repeat !important;
+    }
+
+  } // body.guide
+
+  a:link, a:visited {
+    background: transparent;
+    font-weight: bold;
+    text-decoration: underline;
+    }
```
[the remark this answers is not in the document]
Comment (carlosantoniodasilva) [reply guides/assets/stylesrc/print.scss:177]: Also, maybe move these link definitions up with the others that are near the beginning, just to keep them close together. (lines 21-24)

**Review by carlosantoniodasilva — COMMENTED** [review]
@jathayde nice! it's great to have some good print styles for these.

I had a few bits of feedback below, most importantly the previous print.css source file that we need to remove.

I was also wondering if we had any control over the page margin, it seems like the top could get some extra space, see the screenshots below.

<details><summary>Margin Samples</summary>
<p>

Screenshot 2024-04-09 at 11 00 23
Screenshot 2024-04-09 at 11 00 34
Screenshot 2024-04-09 at 11 00 30

</p>
</details> 

Totally cool to leave it if that's not something we have much control over, it was just something I noticed on a quick scan through the printed PDF.

**Comment by jathayde** [comment]
Setting the `@page` element and defining the margins explicitly (5f06956b):

Screenshot 2024-04-16 at 9 52 46 AM

Explicitly defining `pre` font size for code blocks to make it feel more inline with the body (9pt in IBM Plex Mono vs 10pt in Inter body) (61132c72)

Screenshot 2024-04-16 at 9 54 56 AM

Removed static print.css in e5e7beca

Fixed anchor considerations in b69c5c4a

**Review by carlosantoniodasilva — COMMENTED** [review]
Thanks for the updates @jathayde! I think the print styles are looking pretty great now, there's just one thing that is not sitting super right: the inner code pieces still seem to have bigger font-size, causing them to standout a little too much.

<details><summary>Here's a few examples</summary>
<p>

Screenshot 2024-04-17 at 14 09 07

Screenshot 2024-04-17 at 14 10 23

Screenshot 2024-04-17 at 14 10 40

Screenshot 2024-04-17 at 14 10 44

</p>
</details> 

Even some of the code blocks get parts of the code with different font-size, I'm not sure exactly which element is that, but we might need to target it somehow to adjust the font-size accordingly, otherwise it feels there's too much variation between normal text, inner code highlight, and some of the elements in code blocks. I hope the screenshots can show that, but let me know if you need more info. Thanks!

**Comment by jathayde** [comment]
Ah this is a cascade failure. There's a separate call for inline stuff that's throwing this. Let me get that change in.

**Comment by jathayde** [comment]
Screenshot 2024-04-18 at 8 12 37 AM
Screenshot 2024-04-18 at 8 13 59 AM

@carlosantoniodasilva I've moved the `pre` and `code` calls globally in print.css to 9.5, and then also they exist under the `.code` interstitial call as well (to override the original css calls)

**Comment by carlosantoniodasilva** [comment]
@jathayde thanks! The updates are looking great.

I noticed one more minor thing on a quick final pass, that I wanted to hear your thoughts whether it's something worth looking into / potentially changing.

Code inline elements that are also links appear to be super bold, where they also standout a bit too much, see the screenshots for reference:

<details><summary>Details</summary>
<p>

Screenshot 2024-04-18 at 10 13 47

Screenshot 2024-04-18 at 10 14 17

Screenshot 2024-04-18 at 10 14 24

</p>
</details> 

If you think it makes sense (and it's possible) to tweak that a bit, maybe decrease the bold some, then let's do it, other than that this LGTM. Thanks a bunch!

**Comment by carlosantoniodasilva** [comment]
Ah I should've mentioned I was testing on Safari, but here's a Chrome vs Safari comparison for reference (before the last change):

<details><summary>Chrome vs Safari on Mac (before)</summary>
<p>

Screenshot 2024-04-18 at 13 23 32

</p>
</details> 

Chrome does look nicer than Safari with bold there.

Here's the vs after the latest change:

<details><summary>Chrome vs Safari on Mac (after)</summary>
<p>

Screenshot 2024-04-18 at 13 24 54

</p>
</details> 

The updated version looks great to me! Thanks.

**Pull request merged**